sdk/eventhub/Microsoft.Azure.EventHubs/src/Amqp/AmqpPartitionReceiver.cs

```diff
@@ -104,15 +104,15 @@ protected override async Task<IList<EventData>> OnReceiveAsync(int maxMessageCou
                 {
                     // Evaluate retry condition?
                     TimeSpan? retryInterval = this.RetryPolicy.GetNextRetryInterval(ex, timeoutHelper.RemainingTime(), ++retryCount);
-                    if (retryInterval != null && !this.EventHubClient.IsClosed)
+                    if (retryInterval != null && !this.IsClosed)
```

Comment:
Does this change the behavior from a customer's perspective?   If I had code that assumed closing the `EventHubClient` would  cascade to senders/receivers, would I see a different result now, or was that not previously a valid assumption for me to hold?

Reply:
The cascade happens via a completely separate path. This change appears to handle the scenario where the receiver has been closed but the underlying EventHubClient is still open. It might be worth checking both conditions -- is it possible to race between here and the cascade when an EventHubClient closes?

Reply:
Parent client closes all child entities on Close call. This thing I am fixing actually was a bug. W/o fix, we don't observe child entity only close cases.

Reply:
Updated the code to check parent client's close state as well.